### PR TITLE
PLAT-708: test finish partial object deactivation

### PR DESCRIPTION
**Pull request by MikhailZhukov**
Test finish partial object deactivation

### CODE REVIEW

File: ledger-core/virtual/integration/deactivate_test.go [hunk 1]
```diff
@@ -746,3 +746,165 @@ func TestVirtual_CallMethod_After_Deactivation(t *testing.T) {
...
+	for _, testCase := range cases {
+		t.Run(testCase.name, func(t *testing.T) {
+			mc := minimock.NewController(t)
+
+			server, ctx := utils.NewUninitializedServer(nil, t)
+
+			oneExecutionEnded := server.Journal.WaitStopOf(&execute.SMExecute{}, 1)
+
+			runnerMock := logicless.NewServiceMock(ctx, mc, func(execution execution.Context) string {
+				return execution.Request.CallSiteMethod
```
Comment (ElSacramento) [inline ledger-core/virtual/integration/deactivate_test.go:780]: You can use default mock

File: ledger-core/virtual/integration/deactivate_test.go [hunk 2]
```diff
@@ -746,3 +746,165 @@ func TestVirtual_CallMethod_After_Deactivation(t *testing.T) {
...
+
+			server.IncrementPulseAndWaitIdle(ctx)
+			outgoing := server.BuildRandomOutgoingWithPulse()
+			checkOutgoing := server.BuildRandomOutgoingWithPulse()
+
+			{
+				runnerMock.AddExecutionClassify(checkMethod, testCase.isolation, nil)
+				if testCase.isolation.State == contract.CallValidated {
+					runnerMock.AddExecutionMock(checkMethod).AddStart(
+						nil,
```
Comment (ElSacramento) [inline ledger-core/virtual/integration/deactivate_test.go:804]: check that it's validated memory, please

File: ledger-core/virtual/integration/deactivate_test.go (hunk 1 above)
Comment (MikhailZhukov) [reply ledger-core/virtual/integration/deactivate_test.go]: done

File: ledger-core/virtual/integration/deactivate_test.go (hunk 2 above)
Comment (MikhailZhukov) [reply ledger-core/virtual/integration/deactivate_test.go]: done

File: ledger-core/virtual/integration/deactivate_test.go [hunk 3]
```diff
@@ -746,3 +746,165 @@ func TestVirtual_CallMethod_After_Deactivation(t *testing.T) {
...
+				require.Equal(t, objectRef, report.Object)
+				require.Equal(t, payload.Inactive, report.Status)
+				require.True(t, report.DelegationSpec.IsZero())
+				require.Equal(t, int32(0), report.UnorderedPendingCount)
+				require.Equal(t, int32(0), report.OrderedPendingCount)
+				require.Nil(t, report.ProvidedContent)
+				return false
+			})
+			typedChecker.VDelegatedCallResponse.Set(func(response *payload.VDelegatedCallResponse) bool {
+				// send delegation request finished with deactivate flag
```
Comment (ElSacramento) [inline ledger-core/virtual/integration/deactivate_test.go:835]: Don't we need to check VDelegatedCallResponse?

Comment (MikhailZhukov) [reply ledger-core/virtual/integration/deactivate_test.go:946]: add some checks

File: ledger-core/virtual/integration/deactivate_test.go [hunk 4]
```diff
@@ -852,3 +852,174 @@ func TestVirtual_DeduplicateCallAfterDeactivation_PrevVE(t *testing.T) {
...
+					Callee:              objectRef,
+					CallSiteDeclaration: class,
+					CallSiteMethod:      "Check",
+					CallOutgoing:        checkOutgoing,
+				}
+				server.SendPayload(ctx, &pl)
+			}
+			commonTestUtils.WaitSignalsTimed(t, 10*time.Second, oneExecutionEnded)
+			server.IncrementPulseAndWaitIdle(ctx)
+			commonTestUtils.WaitSignalsTimed(t, 10*time.Second, gotResult)
```
Comment (ElSacramento) [inline ledger-core/virtual/integration/deactivate_test.go:1014]: Not custom channel, use `commontestutils.WaitSignalsTimed(t, 10*time.Second, typedChecker.VStateReport.Wait(ctx, 1))` instead

Comment (MikhailZhukov) [reply ledger-core/virtual/integration/deactivate_test.go]: thanks a lot, done

**Pull request merged**